app/src/main/java/com/github/fribourgsdp/radio/data/view/PlaylistSongsFragment.kt

```diff
@@ -69,35 +72,59 @@ open class PlaylistSongsFragment : MyFragment(R.layout.fragment_playlist_display
         deleteButton = requireView().findViewById(R.id.deleteButton)
         deleteButton.setOnClickListener{
             //removes playlist from user playlists
-            user.removePlaylist(playlist)
-            user.save(requireContext())
-            user.onlineCopyAndSave()
-            activity?.setResult(0)
-            activity?.finish()
+
+            if(!hasConnectivity(requireContext()) && playlist.savedOnline){
+                disableButtons()
+                Toast.makeText(requireContext(),getString(R.string.offline_error_message_toast), Toast.LENGTH_SHORT).show()
+            }else{
+                user.removePlaylist(playlist)
```

Comment:
Here since you checked for connectivity, could you not re-enable the buttons? Like in the case where you start this fragment and you have no connection, they get disabled, but on pressing delete, since you check again, you may as well allow the user to click these buttons again.